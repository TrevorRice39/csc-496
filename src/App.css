.App {
  text-align: center;
  font-family: Georgia, 'Times New Roman', Times, serif
}

.App-logo {
  height: 10vmin;
}

/*
#EC91D8
#FFAAEA
#FFEED6
#A77E58
#FBBFCA
*/

.App-header {
  background-color: #BF936B;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  text-align: left;
  padding-left: 15px;
  padding-top: 20px;;
  padding-bottom: 10px;
  
  justify-content: center;
  font-size: calc(20px + 2vmin);
  
  color: black;
}

.App-link {
  color: #09d3ac;
}
body {
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: rgb(83, 80, 80);
  min-height: 80vh;
  
  background-color: #FFEED6;
  /* background-color: #FBBFCA; */
}
footer {
  background-color: #A77E58;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
  font-size: calc(2px + 2vmin);
  color:black;
}

.nav_button {
  width: 125px;
  outline: 0;
  border: none;
  font-size: 16px;
  margin-top: 0px;
  margin-bottom: 15px;
  padding: 12px 16px ;
  border-radius: 2.5%;
  background-color:#FFAAEA;
  box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);
}
.nav_button::-moz-focus-inner {
  border: 0;
}
.nav_button:hover {
  border: 0;
  border-radius: 5%;
  background-color: #EC91D8;
}

h2 {
  text-align: left;
  margin: 20px;
}
.recipe_para {
  text-align: left;
  margin: 20px;
}

img {
  padding: 15px;
  border-radius: 8px;
  border: rgb(204, 13, 13);
  
}
img:hover {
  box-shadow: 0 5px 5px 0 rgba(0,0,0,0.24), 0 10px 10px 0 rgba(0,0,0,0.19);
}