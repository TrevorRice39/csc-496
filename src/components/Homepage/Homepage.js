import React, { Component } from "react";
import RecipeSummaries from './RecipeSummaries/RecipeSummaries.js';
import MainContent from './MainContent/MainContent.js';

class Homepage extends Component {
    render() {
      const {recipes} = this.props;
      return (
        <div>
        <h1>Homepage</h1>
          <MainContent/>
          <RecipeSummaries recipes = {recipes}/>
        </div>
      )
    }
  }
  
  export default Homepage;