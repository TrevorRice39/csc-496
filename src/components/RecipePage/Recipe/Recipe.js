import React, { Component } from "react";

class Recipe extends Component {
  
    constructor(props) {
      super(props);
  
      //this.getImages = this.getImages.bind(this);
    }
  
    render() {
      const {recipe} = this.props;
      const images = getImages(recipe.field_images);
      return (
        <div>
          {typeof recipe != 'undefined'?
          <body>
            <h1>{recipe.title}</h1> 
            {typeof images !== 'undefined'?
              images.map(url =>
                <img src = {"http://gtest.dev.wwbtc.com" + url}></img>
              )
              : <div></div>
            }
            <h2>Preparation</h2>
            <p className = "recipe_para" dangerouslySetInnerHTML={{ __html: recipe.body}}></p>
            <h2>Ingredients</h2>
            <p className = "recipe_para">{recipe.field_ingredients}</p>
            
          </body>
            : <div></div>
            }
        </div>
      )
    }
}
const getImages = (urls_string) => {
    var urls = [];

    var blankIndex = urls_string.indexOf(' ');
    while (blankIndex != -1) {
      var url = urls_string.substring(0, blankIndex-1);
      urls.push(url);
      urls_string = urls_string.substring(blankIndex+1);
      blankIndex = urls_string.indexOf(' ');
    }

    if (urls_string.length > 0) {
      urls.push(urls_string);
    }
    return urls;
}
export {
    getImages,
}
export default Recipe;