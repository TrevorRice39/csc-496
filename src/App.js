import React, { Component } from "react";
//import logo from "./recipes_logo.jpg";
import "./App.css";

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      recipes: [],
      homepage: true,
    };
  }
  render() {
    const {recipes, homepage} = this.state;
    return (
      <div className="App">
        <header className="App-header">
          Trevor's Recipes
        </header>
        <nav>
          {/* homepage button
              recipe button
          */}
        </nav>
        <body>
          <h1>Homepage</h1>
        {homepage?
        <div>
        <MainContent/>
        <RecipeSummaries recipes = {recipes}/>
        </div>
        : <h1>boy this the recipe page</h1>
        }
        </body>
        <footer>
          Trevor Rice
          <br></br>
          CSC 496
        </footer>
      </div>
    );
  }

  componentDidMount() {
    fetch("http://gtest.dev.wwbtc.com/json/rec")
      .then(response => response.json())
      .then(result => this.setState({recipes: result, loading: false}))
      .catch(error => console.log(error));

  }
}

class MainContent extends Component {
  
  constructor(props) {
    super(props);

    this.state = { 
      homepage: [], 
    }
  }
  render() {
    const {homepage} = this.state;
    return (
      <div>
        {homepage.map(res =>
          <div dangerouslySetInnerHTML={{ __html: res.body }} />
          )}
        </div>
    )
  }
  componentDidMount() {
    fetch("http://gtest.dev.wwbtc.com/json/page?_format=json")
      .then(response => response.json())
      .then(result => this.setState({homepage: result}))
      .catch(error => console.log(error));
  }
}


class RecipeSummaries extends Component {
  
  render() {
    const {recipes} = this.props;
    console.log(recipes);
    return (
      <div>
        <h1>A few recipes!</h1>
        {recipes.map(recipe =>

          <div>
            
          {recipe.title}
          :&nbsp;
          <div dangerouslySetInnerHTML={{ __html: recipe.field_summary }} />
          </div>
          )}
      </div>
    )
  }
}

/*
{recipes.map(recipe =>
        <div>
          <h12>{recipe.title}</h12>
          <div dangerouslySetInnerHTML={{ __html: recipe.body }} />
          <img src = {"http://gtest.dev.wwbtc.com" + recipe.field_images.substring(0, recipe.field_images.indexOf(','))}></img>
          </div>
        )} 
*/
export default App;
