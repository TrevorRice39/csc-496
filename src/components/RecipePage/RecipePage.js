import React, { Component } from "react";
import Recipe from './Recipe/Recipe.js';
import {Button} from '../../App.js';

class RecipePage extends Component {

    constructor(props) {
      super(props);
      
      this.state = {
        recipeIndex: 0,
      }
  
      this.previousRecipe = this.previousRecipe.bind(this);
      this.nextRecipe = this.nextRecipe.bind(this);
    }
  
    nextRecipe() {
      const {recipeIndex} = this.state;
      const {recipes} = this.props;
      if (recipeIndex < recipes.length - 1) {
        this.setState({recipeIndex: recipeIndex+1})
      }
    }
  
    previousRecipe() {
      const {recipeIndex} = this.state;
      if (recipeIndex > 0) {
        this.setState({recipeIndex: recipeIndex-1})
      }
    }
    render() {
      const {recipeIndex} = this.state;
      const {recipes} = this.props;
      return (
        <div>
        <Recipe recipe = {recipes[recipeIndex]}/>
        <Button className = "nav_button" onClick = {() => this.previousRecipe()} text = "<< Previous"/>
        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
        <Button className = "nav_button" onClick = {() => this.nextRecipe()} text = "Next >>"/>
        </div>
      )
    }
  }
export default RecipePage;