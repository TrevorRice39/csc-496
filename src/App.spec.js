import React from 'react';
import renderer from 'react-test-renderer';
import "isomorphic-fetch";
import {mount} from 'enzyme';
import App, {Button} from './App.js';
import Homepage from './components/Homepage/Homepage.js';
import Recipe, {getImages} from './components/RecipePage/Recipe/Recipe.js';
import RecipePage from './components/RecipePage/RecipePage.js';
import { tsExternalModuleReference } from '@babel/types';
import ShallowRenderer from 'react-test-renderer/shallow';



describe('App', () => {
  it('renders the homepage', () => {
    const wrapper = mount(<App homepage = {true}/>);
    expect(wrapper.find(Homepage).length).toEqual(1);
  });
})

// describe('App', () => {
//   it('renders the recipe page', () => {
//     const renderer = new ShallowRenderer();
//     renderer.render(<App homepage = {false}/>);
//     const result = renderer.getRenderOutput();
//     expect(result.type).toBe('div');
//     var recipes = [];
//     var expected = 
//       <div>
//       <header className="App-header">
//           <div>
//           Recipes Bazaar
//           &nbsp;&nbsp;
//           <Button className = "nav_button bouncy" style="animation-delay:0.07s;" onClick = {() => this.switchPage("homepage")} text = "Homepage"/>
//          &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
//          <Button className = "nav_button bouncy" style="animation-delay:1.00s;" onClick = {() => this.switchPage("recipes")} text = "Recipes"/>
//          </div>
//         </header>
//         <div>
//         <body>
//          <RecipePage recipes = {recipes}/>
//          </body>
         
//           <footer>
//           Trevor Rice
//           <br></br>
//           CSC 496
//         </footer>
//         </div>
//         </div>
//         ;
//     expect(JSON.stringify(result.props.children)).toEqual(JSON.stringify(expected))
//   });
// })

describe('Recipe Page', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RecipePage recipes={[]}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
describe('Recipe', () => {
  it('renders the recipe page', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<RecipePage props = {[]} />);
    const result = renderer.getRenderOutput();
    expect(result.type).toBe('div');
    var expected = [
      <Recipe recipe = {undefined}/>,
        <Button className = "nav_button" onClick = {() => this.previousRecipe()} text = "<< Previous"/>,
        <span>       </span>,
        <Button className = "nav_button" onClick = {() => this.nextRecipe()} text = "Next >>"/>
    ]
    expect(JSON.stringify(result.props.children)).toEqual(JSON.stringify(expected))
  });
})
describe('Homepage', () => {
    test('snapshot renders', () => {
      const recipe  = [{title: "title", body: "body", field_images: "image1", view_node: "node", field_ingredients: "ingredient1", field_summary: "summary"}];
      const component = renderer.create(<Homepage recipes = {recipe}/>);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

describe('Recipe', () => {
  describe ('getImages', () => {
    it('should parse urls', () => {
      const url_string = "url1.com, url2.com, url3.com";
      const parsed_urls = getImages(url_string);
      expect(parsed_urls).toEqual(["url1.com", "url2.com", "url3.com"]);
    });
  });
});
