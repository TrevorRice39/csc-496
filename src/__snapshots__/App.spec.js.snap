// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage snapshot renders 1`] = `
<div>
  <h1>
    Homepage
  </h1>
  <div />
  <div>
    <h1>
      A few recipes!
    </h1>
    <div>
      <h4>
        title
        :
      </h4>
      <p
        className="recipe_para"
        dangerouslySetInnerHTML={
          Object {
            "__html": "summary",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Recipe Page snapshot renders 1`] = `
<div>
  <div>
    <div />
  </div>
  <button
    className="nav_button"
    onClick={[Function]}
  >
    &lt;&lt; Previous
  </button>
  <span>
           
  </span>
  <button
    className="nav_button"
    onClick={[Function]}
  >
    Next &gt;&gt;
  </button>
</div>
`;

exports[`Recipe component snapshot renders 1`] = `
<div>
  <div />
</div>
`;
